api/views.py

```diff
@@ -43,4 +45,25 @@ def post(self, request):
                 return Response("The requested resource could not be found but may be available in the future.", status=status.HTTP_404_NOT_FOUND)
         except Exception:
             return Response("Must supply question key", status=status.HTTP_400_BAD_REQUEST)
-        return Response(data)
\ No newline at end of file
+        return Response(data)
+
+
+class LogAttempt(LoginRequiredMixin, APIView):
+    def post(self, request):
+        try:
+            user_answer = request.data['answer']
+            key = request.data['key']
+            user = request.user
+            cached_response = cache.get(key)
+            class_name = cached_response['class_name']
+            correct_answer = cached_response['answer']
+            question_instance = Question.objects.get(class_name=class_name)
+            user_instance = User.objects.get(username=user)
+            if user_answer == correct_answer:
```

Comment:
You can simplify and remove the if/else
`user_correct = bool(user_answer == correct_answer)
Attempt.objects.create(question=question, user=user_instance, correct=user_correct`

Reply:
this is so great